### PR TITLE
1400 - ids-list-view and ids-list-builder refactor for ids-list-view-item

### LINKED ISSUE #1400: IdsListView: Minor Refactor and Fix Search and Selection
**Describe the bug**
After https://github.com/infor-design/enterprise-wc/pull/1377 we left list view in a state that needs some fixes. Namely:

- all the selection examples
- the search example
- additional refactoring https://github.com/infor-design/enterprise-wc/pull/1398

**To Reproduce**
1. on http://localhost:4300/ids-list-view/list-view-items.html i see two selected and selection does not change on click
2. similarly on http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html cant "activate" the rows by clicking
3. http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html also a problem changing selection
4. on http://localhost:4300/ids-list-view/list-view-items-search.html 
  a) the search doesn't seem to work
  b) it would look better if it was 100% width

**Expected behavior**

Cleaner code from a minor refactor and working search and selection

**Additional context**
https://github.com/infor-design/enterprise-wc/pull/1377

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

This PR fixes some outstanding issues in [this older PR](https://github.com/infor-design/enterprise-wc/pull/1377).  
> 1. on http://localhost:4300/ids-list-view/list-view-items.html i see two selected but i cant seem to change selection. Maybe just remove selection from the example page since its on another page?
> 2. similarly on http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html i cant seem to "activate" the rows (change by clicking)
> 3. http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html also a problem changing selection
> 4. on http://localhost:4300/ids-list-view/list-view-items-search.html a) the search doesnt seem to work and b) it would look better if it was 100% width

I also refactored `IdsListView` and `IdsListBuilder` to improve integration with new `IdsListViewItem` child-component.

```html
<ids-list-view selectable="multiple" sortable>
  <ids-list-view-item>
    <ids-text font-size="16" type="h2">First Column</ids-text>
    <ids-text font-size="12" type="span">ID: 1</ids-text>
    <ids-text font-size="12" type="span">Comments: One</ids-text>
  </ids-list-view-item>
  <ids-list-view-item active>
    <ids-text font-size="16" type="h2">Second Column</ids-text>
    <ids-text font-size="12" type="span">ID: 2</ids-text>
    <ids-text font-size="12" type="span">Comments: Two</ids-text>
  </ids-list-view-item>
  <ids-list-view-item disabled>
    <ids-text font-size="16" type="h2">Third Column</ids-text>
    <ids-text font-size="12" type="span">ID: 3</ids-text>
    <ids-text font-size="12" type="span">Comments: Three</ids-text>
  </ids-list-view-item>
</ids-list-view>
```

**Related github/jira issue (required)**:

Closes #1400  

**Steps necessary to review your pull request (required)**:

1. Check out branch and run: `nvm use && npm run start`
2. Check all examples here to ensure no regressions: http://localhost:4300/ids-list-view
4. Check all examples here to ensure no regressions: http://localhost:4300/ids-list-builder
5. Check all examples here to ensure no regressions: http://localhost:4300/ids-swappable
9. Run tests to ensure they all pass: `npx jest -- list-view`

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/ids-list-builder/demos/selection-multiple.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
       <ids-layout-grid-cell>
         <ids-list-builder height="310px" selectable="multiple" id="list-builder-2" hide-checkboxes="true">
           <template>
-            <ids-text font-size="16" type="span">${manufacturerName}</ids-text>
+            <ids-text manufacturerName font-size="16" type="span">${manufacturerName}</ids-text>
```
Comment (tmcconechy) [inline src/components/ids-list-builder/demos/selection-multiple.html:17]: Is this correct? Looks wrong casing / misplaced?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Just see a wierd attribute i think is misplaced. Also a lint error.

In terms of functionality:

- [x] http://localhost:4300/ids-list-view/list-view-items.html -> on this example cant seem to change selection. I also see one that is `active` thats only correct for mixed selection so my want to remove

- [x] http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html -> i think itsnt quite correct either. I can select via checkbox but on that one it should work that you can select all https://main-enterprise.demo.design.infor.com/components/listview/example-multiselect.html we might want to add a setting clickCheckboxToSelect and have both options. But also the activation state like its mixed is not needed here

- [x] http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html -> works mostly correct except more than one can be activated ... see https://main-enterprise.demo.design.infor.com/components/datagrid/example-mixed-selection.html

- [ ] http://localhost:4300/ids-list-view/card.html -> here the pager breaks out of the card noticed it here as well http://localhost:4300/ids-list-view/pagination.html i think it should only span the contents that the grid does with maybe 8px padding

- [x] http://localhost:4300/ids-list-view/embellishment.html -> i think the badge here should be a real badge it looks strange maybe change to a number like the badge examples or use. tag instead

Overall looks good 👍🏻

File: src/components/ids-list-builder/demos/selection-multiple.html (hunk 1 above)
Comment (clayinfor) [reply src/components/ids-list-builder/demos/selection-multiple.html]: Hey @tmcconechy , this has been updated to the following:
```html
<ids-text data-field="manufacturerName" font-size="16" type="span">${manufacturerName}</ids-text>
```

It's used by the `IdsListViewItem.templateElements()` function which is needed in the `IdsListBuilder.edit()` method to determine how to properly edit/update the text in the list-view-item

**Comment by clayinfor** [comment]
hey @tmcconechy , I've adjusted some of the pager-styles, but completely fixing the pager is a can of worms, and I've concluded out of scope for this particular task.  I think it makes more sense to handle it in a pager-centric ticket.

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Pull request merged**